aws/cloudformation/lambdas/sql_user/sql_user.rb

```diff
@@ -0,0 +1,98 @@
+require 'mysql2'
+require 'json'
+require_relative './cfn_response'
+
+def lambda_handler(event:, context:)
+  props = event['ResourceProperties']
+  name = props['Name']
+  client_host = props['ClientHost']
+  password = props['Password']
+  databases = props['Databases']
+  privileges = props['Privileges']
+
+  physical_resource_id = event['PhysicalResourceId']
+  physical_resource_id ||= username
+
+  mysql_client = Mysql2::Client.new(
+    host: props['DBServerHost'],
+    username: props['DBAdminUsername'],
+    password: props['DBAdminPassword'],
+    read_timeout: 10,
+    write_timeout: 10,
+    connect_timeout: 10
+  )
+
+  case event['RequestType']
+  when 'Create'
+    create_or_update_sql_user(
+      mysql_client: mysql_client,
+      name: name,
+      client_host: client_host,
+      password: password,
+      databases: databases,
+      privileges: privileges
+    )
+  when 'Update'
+    create_or_update_sql_user(
+      mysql_client: mysql_client,
+      name: name,
+      client_host: client_host,
+      password: password,
+      databases: databases,
+      privileges: privileges
+    )
```

Comment:
It looks like the 'create' and 'update' cases are identical, so we should compact them

```suggestion
  when 'Create', 'Update'
    create_or_update_sql_user(
      mysql_client: mysql_client,
      name: name,
      client_host: client_host,
      password: password,
      databases: databases,
      privileges: privileges
    )
```